exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/FederationGatewayRestClientConfiguration.java

```diff
@@ -0,0 +1,101 @@
+package fi.thl.covid19.exposurenotification;
+
+import fi.thl.covid19.exposurenotification.error.RestTemplateErrorHandler;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.PrivateKeyStrategy;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.util.ResourceUtils;
+import org.springframework.web.client.RestTemplate;
+
+import javax.net.ssl.SSLContext;
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.time.Duration;
+
+@Configuration
+public class FederationGatewayRestClientConfiguration {
+
+    private final Duration connectTimeout = Duration.ofSeconds(60);
+    private final Duration readTimeout = Duration.ofSeconds(60);
+    private final Duration socketTimeout = Duration.ofSeconds(60);
+
+    private final String trustStorePath;
+    private final char[] trustStorePassword;
+    private final String keyStorePath;
+    private final char[] keyStorePassword;
+    private final String keyStoreKeyAlias;
+
+    public FederationGatewayRestClientConfiguration(
+            @Value("${covid19.federation-gateway.trust-store.path}") String trustStorePath,
```

Comment:
Actually it would be good to have properties class for properties, but there was some discussion about that is  not good somwhow, so now it's in that way.